### ISSUE TITLE
[Bug]: Once the Boolean Expression has more than 49 conditions, an exception will be reported

### ISSUE THREAD

**Issue by Richard-lrg** [milvus-io/milvus#25848]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.11
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): attu data query
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Once the Boolean Expression has more than 49 conditions, an exception will be reported

expr：
(test["907"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test[""]) == "1" || (test["2"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1")

err：
fail to query on all shard leaders, err=All attempts results: attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=stream operation failed: [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 attempt #2:context canceled

### Milvus Log

[2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/cgo_helper.go:56] ["Create retrieve plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
"]

[2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/task_scheduler.go:338] ["[UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535"]

[2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/cgo_helper.go:56] ["Create retrieve plan by expr failed, C Runtime Exception: [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
"]

[2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/task_scheduler.go:338] ["[UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535"]

[2023/07/21 11:49:07.250 +00:00] [ERROR] [querynode/shard_cluster.go:892] ["Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>"] [stack="github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Query
 \t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:892
github.com/milvus-io/milvus/internal/querynode.(*QueryNode).queryWithDmlChannel
\t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:1117
github.com/milvus-io/milvus/internal/querynode.(*QueryNode).Query.func1
\t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:1192
golang.org/x/sync/errgroup.(*Group).Go.func1
\t/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:7"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [querynode/impl.go:1119] ["failed to query cluste"] [traceID=9e582a43ec7c6e7] [msgID=443007669881536513] [collectionID=442965392018551674] [error="Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [retry/retry.go:44] ["retry func faile"] ["retry time"=0] [error="Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_query.go:491] ["QueryNode query result erro"] [traceID=9e582a43ec7c6e7] [msgID=443007669881536513] [nodeID=58] [reason="Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:133] ["failed to do query with nod"] [traceID=9e582a43ec7c6e7] [nodeID=58] [dmlChannels="[by-dev-rootcoord-dml_4_442965392018551674v0"] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:160] ["retry another query node with round robi"] [traceID=9e582a43ec7c6e7] [Nexts="{\"by-dev-rootcoord-dml_4_442965392018551674v0\":-1"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:60] ["no shard leaders were availabl"] [traceID=9e582a43ec7c6e7] [channel=by-dev-rootcoord-dml_4_442965392018551674v0] [leaders="[<NodeID: 58>"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:120] ["failed to search/query with round-robin polic"] [traceID=9e582a43ec7c6e7] [error="Channel: by-dev-rootcoord-dml_4_442965392018551674v0 returns err: code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_query.go:379] ["invalid shard leaders cache, updating shardleader caches and retry quer"] [traceID=9e582a43ec7c6e7] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [retry/retry.go:44] ["retry func faile"] ["retry time"=0] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>"]

[2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_scheduler.go:473] ["Failed to execute task:"] [error="fail to query on all shard leaders, err=All attempts results:
attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>)
 attempt #2:context canceled"] [traceID=9e582a43ec7c6e7]

[2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/impl.go:3361] ["Query failed to WaitToFinis"] [traceID=9e582a43ec7c6e7] [error="fail to query on all shard leaders, err=All attempts results:
attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert \"unsupported expr proto node\" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535
 err %!w(<nil>)
 attempt #2:context canceled"] [role=proxy] [msgID=443007669881536513] [db=default] [collection=wangshuxun1_test] [partitions="["]

**Comment by xiaofan-luan** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Environment
> ```
> - Milvus version:2.2.11
> - Deployment mode(standalone or cluster): standalone
> - MQ type(rocksmq, pulsar or kafka):    
> - SDK version(e.g. pymilvus v2.0.0rc2): attu data query
> - OS(Ubuntu or CentOS): 
> - CPU/Memory: 
> - GPU: 
> - Others:
> ```
> 
> ### Current Behavior
> Once the Boolean Expression has more than 49 conditions, an exception will be reported
> 
> expr： (test["907"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test[""]) == "1" || (test["2"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1")
> 
> err： fail to query on all shard leaders, err=All attempts results: attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=stream operation failed: [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 attempt #2:context canceled
> 
> ### Expected Behavior
> _No response_
> 
> ### Steps To Reproduce
> _No response_
> 
> ### Milvus Log
> [2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/cgo_helper.go:56] ["Create retrieve plan by expr failed, C Runtime Exception: [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 "]
> 
> [2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/task_scheduler.go:338] ["[UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535"]
> 
> [2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/cgo_helper.go:56] ["Create retrieve plan by expr failed, C Runtime Exception: [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 "]
> 
> [2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/task_scheduler.go:338] ["[UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535"]
> 
> [2023/07/21 11:49:07.250 +00:00] [ERROR] [querynode/shard_cluster.go:892] ["Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("] [stack="github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Query \t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:892 github.com/milvus-io/milvus/internal/querynode.(*QueryNode).queryWithDmlChannel \t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:1117 github.com/milvus-io/milvus/internal/querynode.(*QueryNode).Query.func1 \t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:1192 golang.org/x/sync/errgroup.(*Group).Go.func1 \t/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:7"]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [querynode/impl.go:1119] ["failed to query cluste"] [traceID=9e582a43ec7c6e7] [msgID=443007669881536513] [collectionID=442965392018551674] [error="Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [retry/retry.go:44] ["retry func faile"] ["retry time"=0] [error="Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_query.go:491] ["QueryNode query result erro"] [traceID=9e582a43ec7c6e7] [msgID=443007669881536513] [nodeID=58] [reason="Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:133] ["failed to do query with nod"] [traceID=9e582a43ec7c6e7] [nodeID=58] [dmlChannels="[by-dev-rootcoord-dml_4_442965392018551674v0"] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:160] ["retry another query node with round robi"] [traceID=9e582a43ec7c6e7] [Nexts="{"by-dev-rootcoord-dml_4_442965392018551674v0":-1"]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:60] ["no shard leaders were availabl"] [traceID=9e582a43ec7c6e7] [channel=by-dev-rootcoord-dml_4_442965392018551674v0] [leaders="[<NodeID: 58>"]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:120] ["failed to search/query with round-robin polic"] [traceID=9e582a43ec7c6e7] [error="Channel: by-dev-rootcoord-dml_4_442965392018551674v0 returns err: code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_query.go:379] ["invalid shard leaders cache, updating shardleader caches and retry quer"] [traceID=9e582a43ec7c6e7] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [retry/retry.go:44] ["retry func faile"] ["retry time"=0] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_scheduler.go:473] ["Failed to execute task:"] [error="fail to query on all shard leaders, err=All attempts results: attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w() attempt #2:context canceled"] [traceID=9e582a43ec7c6e7]
> 
> [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/impl.go:3361] ["Query failed to WaitToFinis"] [traceID=9e582a43ec7c6e7] [error="fail to query on all shard leaders, err=All attempts results: attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w() attempt #2:context canceled"] [role=proxy] [msgID=443007669881536513] [db=default] [collection=wangshuxun1_test] [partitions="["]
> 
> ### Anything else?
> _No response_

This is gonna to be very slow even if it could work. What's the use cases of this expression?

**Comment by Cactus-L** [comment]
> > ### Is there an existing issue for this?
> > 
> > * [x]   I have searched the existing issues
> > 
> > ### Environment
> > ```
> > - Milvus version:2.2.11
> > - Deployment mode(standalone or cluster): standalone
> > - MQ type(rocksmq, pulsar or kafka):    
> > - SDK version(e.g. pymilvus v2.0.0rc2): attu data query
> > - OS(Ubuntu or CentOS): 
> > - CPU/Memory: 
> > - GPU: 
> > - Others:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ### Current Behavior
> > Once the Boolean Expression has more than 49 conditions, an exception will be reported
> > expr： (test["907"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test[""]) == "1" || (test["2"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["4"] == "1") || (test["2"] == "1") || (test["1"] == "1") || (test["3"] == "1") || (test["4"] == "1") || (test["1"] == "1") || (test["2"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1") || (test["3"] == "1") || (test["2"] == "1")
> > err： fail to query on all shard leaders, err=All attempts results: attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=stream operation failed: [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 attempt #2:context canceled
> > ### Expected Behavior
> > _No response_
> > ### Steps To Reproduce
> > _No response_
> > ### Milvus Log
> > [2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/cgo_helper.go:56] ["Create retrieve plan by expr failed, C Runtime Exception: [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 "]
> > [2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/task_scheduler.go:338] ["[UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535"]
> > [2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/cgo_helper.go:56] ["Create retrieve plan by expr failed, C Runtime Exception: [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 "]
> > [2023/07/21 11:49:07.249 +00:00] [WARN] [querynode/task_scheduler.go:338] ["[UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535"]
> > [2023/07/21 11:49:07.250 +00:00] [ERROR] [querynode/shard_cluster.go:892] ["Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("] [stack="github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Query \t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:892 github.com/milvus-io/milvus/internal/querynode.(*QueryNode).queryWithDmlChannel \t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:1117 github.com/milvus-io/milvus/internal/querynode.(*QueryNode).Query.func1 \t/go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:1192 golang.org/x/sync/errgroup.(*Group).Go.func1 \t/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:7"]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [querynode/impl.go:1119] ["failed to query cluste"] [traceID=9e582a43ec7c6e7] [msgID=443007669881536513] [collectionID=442965392018551674] [error="Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [retry/retry.go:44] ["retry func faile"] ["retry time"=0] [error="Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_query.go:491] ["QueryNode query result erro"] [traceID=9e582a43ec7c6e7] [msgID=443007669881536513] [nodeID=58] [reason="Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:133] ["failed to do query with nod"] [traceID=9e582a43ec7c6e7] [nodeID=58] [dmlChannels="[by-dev-rootcoord-dml_4_442965392018551674v0"] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:160] ["retry another query node with round robi"] [traceID=9e582a43ec7c6e7] [Nexts="{"by-dev-rootcoord-dml_4_442965392018551674v0":-1"]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:60] ["no shard leaders were availabl"] [traceID=9e582a43ec7c6e7] [channel=by-dev-rootcoord-dml_4_442965392018551674v0] [leaders="[<NodeID: 58>"]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_policies.go:120] ["failed to search/query with round-robin polic"] [traceID=9e582a43ec7c6e7] [error="Channel: by-dev-rootcoord-dml_4_442965392018551674v0 returns err: code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_query.go:379] ["invalid shard leaders cache, updating shardleader caches and retry quer"] [traceID=9e582a43ec7c6e7] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [retry/retry.go:44] ["retry func faile"] ["retry time"=0] [error="code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w("]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/task_scheduler.go:473] ["Failed to execute task:"] [error="fail to query on all shard leaders, err=All attempts results: attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w() attempt #2:context canceled"] [traceID=9e582a43ec7c6e7]
> > [2023/07/21 11:49:07.250 +00:00] [WARN] [proxy/impl.go:3361] ["Query failed to WaitToFinis"] [traceID=9e582a43ec7c6e7] [error="fail to query on all shard leaders, err=All attempts results: attempt #1:code: UnexpectedError, error: fail to Query, QueryNode ID = 58, reason=Query 58 failed, reason [UnexpectedError] Assert "unsupported expr proto node" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/PlanProto.cpp:535 err %!w() attempt #2:context canceled"] [role=proxy] [msgID=443007669881536513] [db=default] [collection=wangshuxun1_test] [partitions="["]
> > ### Anything else?
> > _No response_
> 
> This is gonna to be very slow even if it could work. What's the use cases of this expression?

This is actually an alternative to list in list filtering. If list in list is supported, it is not needed

**Comment by yanliang567** [comment]
@czs007 any updates

**Comment by xiaofan-luan** [comment]
/assign @zhagnlu 

this might be related to your refactor

**Comment by xiaofan-luan** [comment]
keep it for @zhagnlu

**Comment by RudiBrainy** [comment]
Hi, I wanted to ask if there was any news about this problem or if there was another way to make this work.
Thanks

**Comment by yanliang567** [comment]
@zhagnlu any updates ?

**Comment by xiaofan-luan** [comment]
thought this is fixed in latest 2.3 or 2.4

**Comment by xiaofan-luan** [comment]
and usually in is more efficient than a lot of &&

**Comment by zhagnlu** [comment]
Yes, it is fixed in both 2.3 and 2.4, and add case in ci test.

**Comment by yanliang567** [comment]
the fix on query path is done in pr #30021 
@zhagnlu please add a fix on search path as well.

**Comment by caozhuozi** [comment]
@yanliang567  I don't this is the root cause. We are using `v2.3.5`(I think the version does not include the PR you mentioned),  the same filter expression works for the `query` API but not  `search`.
